### PR TITLE
Add bot support for code coverage (codecov)

### PR DESCRIPTION
**Pull request by araspitzu**
This PR adds support for the scala test coverage tool ([scoverage](http://scoverage.org/)), it will help providing info on how much the coverage changes for each pull request, also a badge indicating the current coverage is added to the readme.

### CODE REVIEW

**Comment by t-bast** [comment]
LGTM

One thing that I usually do with codecov is provide a `.codecov.yml` configuration file (at the root of the repo). Codecov's default configuration expects unreasonable code coverage percentage which makes the status check fail: this doesn't prevent you from merging, but when looking at the commit history it will look like the commit build failed (you'll see a big red failure cross). To prevent that I usually set reasonable expectations in that custom configuration.

But that's definitely something that can be added afterwards, once we get tired of seeing failed status checks :)

**Comment by araspitzu** [comment]
Currently scoverage is configured to pass the build with any coverage value (travis run `mvn scoverage:report`), so it will just let you know if it's decreasing but it shouldn't fail the build.

PS. example report [here](https://codecov.io/gh/acinq/eclair/branch/scoverage-build-plugin)

File: .travis.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ before_install:
  - export M2_HOME=$PWD/apache-maven-3.6.0
  - export PATH=$M2_HOME/bin:$PATH
 script:
- - mvn install
+ - mvn scoverage:report
```
Comment (pm47) [inline .travis.yml]: Does this do everything that `mvn install` does? (compile+run tests)

Comment (pm47) [reply .travis.yml]: It looks like it does (https://scoverage.github.io/scoverage-maven-plugin/1.4.0-M5/plugin-info.html).

**Pull request merged**